maskrcnn_benchmark/config/paths_catalog.py

```diff
@@ -6,38 +6,86 @@
-        "voc_2012_train": ("voc/VOC2012", 'train'),
-        "voc_2012_trainval": ("voc/VOC2012", 'trainval'),
-        "voc_2012_val": ("voc/VOC2012", 'val'),
-        "voc_2012_test": ("voc/VOC2012", 'test'),
+        "coco_2014_train": {
+            "img_dir": "coco/train2014",
+            "ann_file": "coco/annotations/instances_train2014.json"
+        },
+        "coco_2014_val": {
+            "img_dir": "coco/val2014",
+            "ann_file": "coco/annotations/instances_val2014.json"
+        },
+        "coco_2014_minival": {
+            "img_dir": "coco/val2014",
+            "ann_file": "coco/annotations/instances_minival2014.json"
+        },
+        "coco_2014_valminusminival": {
+            "img_dir": "coco/val2014",
+            "ann_file": "coco/annotations/instances_valminusminival2014.json"
+        },
+        "voc_2007_train": {
+            "data_dir": "voc/VOC2007",
+            "img_dir": "voc/VOC2007/JPEGImages",
+            "ann_file": "voc/VOC2007/Annotations/pascal_train2007.json",
+            "split": "train"
+        },
+        "voc_2007_val": {
+            "data_dir": "voc/VOC2007",
+            "img_dir": "voc/VOC2007/JPEGImages",
+            "ann_file": "voc/VOC2007/Annotations/pascal_val2007.json",
+            "split": "val"
+        },
+        "voc_2007_test": {
+            "data_dir": "voc/VOC2007",
+            "img_dir": "voc/VOC2007/JPEGImages",
+            "ann_file": "voc/VOC2007/Annotations/pascal_test2007.json",
+            "split": "test"
+        },
+        "voc_2012_train": {
+            "data_dir": "voc/VOC2012",
+            "img_dir": "voc/VOC2012/JPEGImages",
+            "ann_file": "voc/VOC2012/Annotations/pascal_train2012.json",
+            "split": "train"
+        },
+        "voc_2012_val": {
+            "data_dir": "voc/VOC2012",
+            "img_dir": "voc/VOC2012/JPEGImages",
+            "ann_file": "voc/VOC2012/Annotations/pascal_val2012.json",
+            "split": "val"
+        },
+        "voc_2012_test": {
+            "data_dir": "voc/VOC2012",
+            "img_dir": "voc/VOC2012/JPEGImages",
+            "ann_file": "voc/VOC2012/Annotations/pascal_test2012.json",
+            "split": "test"
+        },
+        "cityscapes_fine_instanceonly_seg_train": {
+            "img_dir": "cityscapes/images",
+            "ann_file": "cityscapes/annotations/instancesonly_filtered_gtFine_train.json"
+        },
+        "cityscapes_fine_instanceonly_seg_val": {
+            "img_dir": "cityscapes/images",
+            "ann_file": "cityscapes/annotations/instancesonly_filtered_gtFine_val.json"
+        },
+        "cityscapes_fine_instanceonly_seg_test": {
+            "img_dir": "cityscapes/images",
+            "ann_file": "cityscapes/annotations/instancesonly_filtered_gtFine_test.json"
+        }
+    }
 
+    coco_style_datasets = {
+        key + "_cocostyle": value for key, value in DATASETS.items() if "coco" not in key
```

Comment:
what about also adding as constraints that there is a `img_dir` and an `ann_file` as fields of those elements?
Because I can imagine that people will want to add new datasets that are not going to be using the coco evaluation API, and this will force it use this API right away.
---

Actually, on a second thought. I think that this hides away from the user what are the datasets that support COCO-style loading and evaluation. We also have repeated entries for cityscapes, which basically do the same thing.
Let's maybe keep things a bit more verbose, but at least clearer on the intent?

So, what about:
- add `voc_xxx` and `voc_xxx_cocostyle`
- add only `cityscapes_xxx_cocostyle` (given that we don't support native loaders / evaluators for now)

Are you ok with that? I think that the amount of code-duplication will be fairly small, given that we will only have a handful of datasets that will be manually implemented.

### Longer term
Have a 100% dataset-agnostic evaluation tool, so that those details can be hidden away from the user.

Reply:
That sounds a good idea. 
I believe things will be more verbose and simple by doing that, and users could understand the meanings easier.

If it do have too many datasets in the future, or we've built the dataset-agonistic evaluation tool,
we could think about the new representation by then.